app/src/main/java/edu/usf/sas/pal/muser/model/AlbumData.kt

```diff
@@ -0,0 +1,20 @@
+package edu.usf.sas.pal.muser.model
+
+import com.simplecity.amp_library.model.Album
+import com.simplecity.amp_library.model.Artist
+
+data class AlbumData
+(val id: Long,
+ val name: String,
+ val artists: List<Artist>,
+ val albumArtistName: String,
+ val year: Int,
+ val numSongs: Int,
+ val numDiscs: Int,
+ val dateAdded: Long,
+ val paths: List<String>
+ ){
+    constructor(album: Album): this(album.id, album.name, album.artists, album.albumArtistName,
```

Comment:
I believe you should be able to get rid of these explicit `constructor`s in the data classes if you're calling exclusively from Kotlin code and using named parameters.

Reply:
Currently, I'm using this constructor to populate the `AlbumData` class and it doesn't use any named parameters. Also, the named parameters are only being used to populate the `UiEvent` class.

Reply:
Ah, ok, thanks. Sorry, I read this too quickly the first time - 👍